### PR TITLE
Add Multivariate Gamma and test

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
This is used in the Wishart probability distribution.

File: mvgamma.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// must be greater than zero, and MvLgamma will return NaN if v < (dim-1)/2.
+//
+// See https://en.wikipedia.org/wiki/Multivariate_gamma_function for more
+// information.
+func MvLgamma(v float64, dim int) float64 {
+	if dim < 1 {
+		panic("mathext: negative dimension")
+	}
+	df := float64(dim)
+	if v < (df-1)/2 {
```
Comment (sbinet) [inline mvgamma.go]: replace `/2` with `*0.5`

File: mvgamma.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		return math.NaN()
+	}
+	ans := df * (df - 1) / 4 * logPi
```
Comment (sbinet) [inline mvgamma.go]: replace `/4` with `* 0.25`

File: mvgamma.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	for i := 1; i <= dim; i++ {
+		lg, _ := math.Lgamma(v + float64(1-i)/2)
```
Comment (sbinet) [inline mvgamma.go]: replace `/2` with `*0.5`

File: mvgamma_test.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	for i, test := range []struct {
+		v   float64
+		dim int
+		ans float64
+	}{
+		{10, 5, 58.893841851237397},
+		{3, 1, 0.69314718055994529},
+	} {
+		ans := MvLgamma(test.v, test.dim)
+		if math.Abs(test.ans-ans) > 1e-14 {
```
Comment (sbinet) [inline mvgamma_test.go:23]: replace with `floats.EqualWithinAbs` ?

File: mvgamma_test.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+			t.Errorf("Case %v. Want %v, got %v.", i, test.ans, ans)
```
Comment (sbinet) [inline mvgamma_test.go]: the usual order (at least in the Go stdlib) is: `got=%v want=%v`.

**Review by sbinet — APPROVED** [review]
LGTM with minor (performance) nits.

File: mvgamma.go (hunk 1 above)
Comment (btracey) [reply mvgamma.go]: Done

File: mvgamma.go (hunk 2 above)
Comment (btracey) [reply mvgamma.go]: Done.

File: mvgamma.go (hunk 3 above)
Comment (btracey) [reply mvgamma.go]: Done.

File: mvgamma_test.go (hunk 4 above)
Comment (btracey) [reply mvgamma_test.go:23]: The floats function is the same code, but with a fast path for equality. I don't think we need that here.

File: mvgamma_test.go (hunk 5 above)
Comment (btracey) [reply mvgamma_test.go]: Done.

**Pull request merged**